rails_app/README.md

```diff
@@ -0,0 +1,166 @@
+
+    bin/rails server
+
+With the worker and Rails API server both running, now a shopping cart can be interacted with. First, see which products
+are available in the database:
+
+    bin/rails runner "puts Product.all.inspect"
+
+A scooter with SKU "1234" and a TV with SKU "2345" are both available. Fetch the current shopping cart for `my-user-id`:
+
+    curl http://127.0.0.1:3000/api/shopping_carts/my-user-id
+
+This should return:
+
+    {"error":"Cart not found"}
+
+This makes sense, no products have been added. Add 3 scooters to the cart:
+
+    curl -X POST http://127.0.0.1:3000/api/shopping_carts/my-user-id/entries \
+         -H "Content-Type: application/json" -d '{ "sku": "1234", "quantity": 3 }'
+
+The result will be JSON showing we've added an entry with ID `1` and some details about the entry. Check the cart again:
+
+    curl http://127.0.0.1:3000/api/shopping_carts/my-user-id
+
+The cart is now present with one entry. The current total can be seen too:
+
+    curl http://127.0.0.1:3000/api/shopping_carts/my-user-id/current_total
+
+This is powered by a Temporal workflow. Getting the cart and total are simply workflow queries.
+
+The workflow can be seen in the UI. If using the local dev server with defaults,
+http://localhost:8233/namespaces/default/workflows/shopping-cart-my-user-id will show this cart. Notice how there is an
+`add_cart_entry` workflow update for the product we added and it had to fetch the product details from the database via
+an activity. Can even use the "Queries" tab to fetch the current cart or total.
+
+Now lets remove the 3 scooters, add just 1 scooter, and 2 TVs:
+
+    curl -X DELETE http://127.0.0.1:3000/api/shopping_carts/my-user-id/entries/1
+    curl -X POST http://127.0.0.1:3000/api/shopping_carts/my-user-id/entries \
+         -H "Content-Type: application/json" -d '{ "sku": "1234", "quantity": 1 }'
+    curl -X POST http://127.0.0.1:3000/api/shopping_carts/my-user-id/entries \
+         -H "Content-Type: application/json" -d '{ "sku": "2345", "quantity": 21 }'
+
+The workflow history in the UI clearly shows the removal and addition of cart items. All of this state is in Temporal,
+there is nothing stored in the database at this time. To complete this cart, call checkout with a payment ID:
+
+    curl -X POST http://127.0.0.1:3000/api/shopping_carts/my-user-id/checkout \
+         -H "Content-Type: application/json" -d '{ "payment_id": "my-payment-id" }'
+
+This will apply payment and persist the order in the database. The result will be a JSON with the completed order. The
+workflow is now complete as can be seen in the UI. The order can be seen from the database:
+
+    bin/rails runner "puts Order.all.inspect"
+
+This is obviously just a demonstration for this sample. A real shopping cart may have many more layers to it. For
+instance, it'd be normal to have a script that looks for open shopping cart workflows with products after some time and
```

Comment:
> to have a script that looks for open shopping cart workflows

I would avoid suggesting the idea of "a script" for this. Most implementers would use a workflow timer for this.

Reply:
Not really suggesting it, I just left off what technology they might want to use to do it. Not trying to get into implementation details of how they want to schedule code.